libunwind/src/CompartmentInfo.hpp

```diff
@@ -51,13 +50,23 @@ struct CompartmentInfo {
     CHERI_DBG("C18N: SET IP: %#p\n", state.pc);
 
     for (size_t i = 0; i < sizeof(state.regs) / sizeof(*state.regs); ++i) {
-      registers.setCapabilityRegister(UNW_ARM64_C19 + i, (pint_t)state.regs[i]);
-      CHERI_DBG("C18N: SET REGISTER: %lu (%s): %#p\n",
-                UNW_ARM64_C19 + i,
-                registers.getRegisterName(UNW_ARM64_C19 + i),
+      int reg_num;
+#ifdef _LIBUNWIND_TARGET_AARCH64
+      reg_num = UNW_ARM64_C19 + i;
+#elif defined(_LIBUNWIND_TARGET_RISCV)
+      static constexpr int callee_saved[] = {
```

Comment:
Might be better to hoist this out, define an equivalent array for Morello, and static assert that it's the same length as state.regs.